### PR TITLE
Remove Calendar type parameter from TimeFormat

### PR DESCRIPTION
**Pull request by Manishearth**
Fixes https://github.com/unicode-org/icu4x/issues/2153

Also moves the time keys over to being non-calendared, and takes the opportunity to rename `icu_datetime::date` to `icu_datetime::input`.

### CODE REVIEW

File: components/datetime/examples/work_log.rs [hunk 1]
```diff
@@ -40,7 +40,7 @@ fn print(_input: &str, _value: Option<usize>) {
 fn main(_argc: isize, _argv: *const *const u8) -> isize {
     icu_benchmark_macros::main_setup!();
 
-    let provider = icu_testdata::get_provider();
+    let provider = icu_testdata::get_smaller_postcard_provider();
```
Comment (sffc) [inline components/datetime/examples/work_log.rs]: Nit: Don't check this in

File: components/datetime/src/any/datetime.rs [hunk 2]
```diff
@@ -15,11 +15,11 @@ use icu_provider::prelude::*;
 use crate::provider::{
     calendar::{
         DatePatternsV1Marker, DateSkeletonPatternsV1Marker, DateSymbolsV1Marker,
-        TimePatternsV1Marker, TimeSymbolsV1Marker,
+        TimeLengthsV1Marker, TimeSymbolsV1Marker,
```
Comment (sffc) [inline components/datetime/src/any/datetime.rs:18]: Nit: We have DatePatternsV1Marker mapping to datelengths@1, too. We should be consistent in the naming, by either keeping the status quo, changing the keys to say "patterns", or changing the structs to say "lengths"

File: ffi/diplomat/cpp/docs/source/datetime_ffi.rst [hunk 3]
```diff
@@ -69,7 +69,7 @@
 
 .. cpp:class:: ICU4XGregorianTimeFormatter
 
-    An ICU4X TimeFormatter object capable of formatting a :cpp:class:`ICU4XGregorianDateTime` as a string, using the Gregorian Calendar.
+    An ICU4X TimeFormatter object capable of formatting a :cpp:class:`ICU4XGregorianDateTime` as a string
```
Comment (sffc) [inline ffi/diplomat/cpp/docs/source/datetime_ffi.rst:72]: Nit: Is the input type still `ICU4XGregorianDateTime`?

Comment (Manishearth) [reply ffi/diplomat/cpp/docs/source/datetime_ffi.rst:72]: Yes, but we can give it more input types in the future

**Review by sffc — APPROVED** [review]

**Pull request merged**